pkg/registry/core/shoot/storage/binding.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener/pkg/apis/core"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/apiserver/pkg/storage"
+	"k8s.io/apiserver/pkg/util/dryrun"
+)
+
+// BindingREST implements a RESTStorage for shoots/binding.
+type BindingREST struct {
+	shootStorage *REST
+}
+
+var _ = rest.NamedCreater(&BindingREST{})
+
+// New returns and instance of Binding
+func (r *BindingREST) New() runtime.Object {
+	return &core.Binding{}
+}
+
+// Create binds a shoot to a seed
+func (r *BindingREST) Create(ctx context.Context, name string, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (out runtime.Object, err error) {
+	binding, ok := obj.(*core.Binding)
+	if !ok {
+		return nil, errors.NewBadRequest(fmt.Sprintf("not a Binding object: %#v", obj))
+	}
+
+	if name != binding.Name {
+		return nil, errors.NewBadRequest(fmt.Sprintf("name in URL does not match name in Binding object: %v,%v", name, binding.Name))
+	}
+
+	if createValidation != nil {
+		if err := createValidation(ctx, binding.DeepCopyObject()); err != nil {
+			return nil, err
+		}
+	}
+
+	shootObj, err := r.shootStorage.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	shoot, ok := shootObj.(*core.Shoot)
+	if !ok {
+		return nil, errors.NewInternalError(fmt.Errorf("cannot convert to *core.Shoot object - got type %T", shootObj))
+	}
+
+	if err := r.assignShoot(ctx, binding.UID, binding.ResourceVersion, shoot.Name, binding.Target.Name, dryrun.IsDryRun(options.DryRun)); err != nil {
+		return nil, fmt.Errorf("failed to create binding: %w", err)
+	}
+
+	out = &metav1.Status{Status: metav1.StatusSuccess}
+	return
+}
+
+func (r *BindingREST) assignShoot(ctx context.Context, shootUID types.UID, shootResourceVersion, shootName string, seedName string, dryRun bool) (err error) {
```

Comment:
I am still confused why you call it `shootUID` here while you pass `binding.UID`. Can you explain? If this is just "copy and paste" then rename it to `bindingUID` here as well.

Reply:
it is called like that because the BindingUID passed by the scheduler is the podUID
https://github.com/kubernetes/kubernetes/blob/b290671b1b28ab4af8164e2b69b334fab584558e/pkg/scheduler/scheduler.go#L471
This, I have to adapt in our scheduler as well.
I still have some confusions left though, which I will add later in the discussion.

Reply:
OK, I think I understand this now. It is to ensure that the created `Binding` is associated with the right shoot (in case it got deleted and recreated with the same name in the meantime).
Let's resolve this comment.

Reply:
Yes.